src/Common/PoolBase.h

```diff
@@ -167,6 +170,29 @@ class PoolBase : private boost::noncopyable
         return items.size();
     }
 
+    /// proton: starts
+    void waitForNoMoreInUse()
```

Comment:
One implementation is just move this function to derived class `ConsumerPool.h` since we have overriden shutdown there, from where `waitForNoMoreInUse()` can be invoked.

Reply:
I thought about that too, the reason why I didn't like it is it requires moving `mutex` and `available` from `private` to `protected` in `PoolBase`. I wanted to keep them as internal details of `PoolBase`. But I am open to make the change, let me know what you think.

Reply:
i c. Let's leave it as it is.